rethinkdb/datadog_checks/rethinkdb/queries.py

```diff
@@ -0,0 +1,234 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Iterator, Mapping, Tuple
+
+import rethinkdb
+
+from .connections import Connection, ConnectionServer
+from .types import (
+    ClusterStats,
+    ConfigTotals,
+    CurrentIssuesTotals,
+    Job,
+    JoinRow,
+    ReplicaStats,
+    Server,
+    ServerStats,
+    ServerStatus,
+    ShardReplica,
+    Table,
+    TableStats,
+    TableStatus,
+)
+
+
+class QueryEngine(object):
+    """
+    Definition of RethinkDB queries used by the RethinkDB check.
+
+    Python ReQL reference documentation: https://rethinkdb.com/api/python/
+    """
+
+    def __init__(self, r=None):
+        # type: (rethinkdb.RethinkDB) -> None
+        self._r = rethinkdb.r if r is None else r
+
+    def query_connected_server_version_string(self, conn):
+        # type: (Connection) -> str
+        """
+        Return the raw string of the RethinkDB version used by the server at the other end of the connection.
+        """
+        r = self._r
+        # See: https://rethinkdb.com/docs/system-tables/#server_status
+        server = conn.server()  # type: ConnectionServer
+        server_status = conn.run(r.db('rethinkdb').table('server_status').get(server['id']))  # type: ServerStatus
```

Comment:
Since we are always using `r.db('rethinkdb')`, maybe we can have a `self._db` instead.

Reply:
Good point. We can actually conncet to a specific database (eg `conn = r.connect(db='rethinkdb')`) so I'll update the place where we create the connection. :-)

Reply:
Addressed in f123755eb